dwave/optimization/src/nodes/mathematical.cpp

```diff
@@ -176,18 +195,92 @@ void BinaryOpNode<BinaryOp>::propagate(State& state) const {
                 changes.emplace_back(index, old, values[index]);
             }
         }
+    } else if (lhs_ptr->size() == 1) {
+        // lhs is a single value being broadcast to the rhs array.
+
+        // Create a unary version of our binary op.
+        const double& lhs = lhs_ptr->view(state).front();
+        auto unary_func = std::bind(func, lhs, std::placeholders::_1);
+
+        if (lhs_ptr->diff(state).size()) {
+            // The lhs has changed, so in this case we're probably changing
+            // everything, so just overwrite the state entirely.
+            auto rhs_view = rhs_ptr->view(state);
+            ptr->assign(rhs_view | std::views::transform(unary_func));
+        } else {
+            // The lhs did not change, so go through the changes on the rhs and apply them
+            auto update_func = [&unary_func](Update update) {
+                if (!update.removed()) update.value = unary_func(update.value);
+                return update;
+            };
+            ptr->update(rhs_ptr->diff(state) | std::views::transform(update_func));
+        }
+    } else if (rhs_ptr->size() == 1) {
+        // rhs is a single value being broadcast to the lhs array
```

Comment:
Totally. I briefly had it that way but I went with this approach for two reasons.
1) `std::bind` should be compiled out with this approach. But if we set it at runtime because on which side then it won't be. Very minor performance hit but :shrug: .
2) I also found that approach harder to read. Obviously subjective.

I am open to switching it back.